# HG changeset patch -- Bitbucket.org
# Project eigen2
# URL http://bitbucket.org/cryos/eigen2/overview/
# User Marcus D. Hanwell <marcus@cryos.org>
# Date 1249148586 14400
# Node ID 3c11267138fe98757d27308d2133d0182da740f3
# Parent 3b8120f077865e2a072e10f5be33e1d942b83a06
Improved quoting of tests when added to the build.

This fixes an issue where multiple versions of the Qt libraries are
available, if the Qt library variable is not quoted an error was
generated as only the first part 'optimized' was used by the create test
macro.

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -158,7 +158,7 @@ macro(ei_add_test testname)
     string(STRIP "${ARGV2}" ARGV2_stripped)
     string(LENGTH "${ARGV2_stripped}" ARGV2_stripped_length)
     if(${ARGV2_stripped_length} GREATER 0)
-      target_link_libraries(${targetname} ${ARGV2})
+      target_link_libraries(${targetname} "${ARGV2}")
     endif(${ARGV2_stripped_length} GREATER 0)
   endif(${ARGC} GREATER 2)
 
@@ -212,7 +212,7 @@ ei_add_test(alignedbox)
 ei_add_test(regression)
 ei_add_test(stdvector)
 if(QT4_FOUND)
-  ei_add_test(qtvector " " ${QT_QTCORE_LIBRARY})
+  ei_add_test(qtvector " " "${QT_QTCORE_LIBRARY}")
 endif(QT4_FOUND)
 ei_add_test(sparse_vector)
 ei_add_test(sparse_basic)
